google/resource_compute_shared_vpc_host.go

```diff
@@ -0,0 +1,91 @@
+package google
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceComputeSharedVpcHost() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceComputeSharedVpcHostCreate,
+		Read:   resourceComputeSharedVpcHostRead,
+		Delete: resourceComputeSharedVpcHostDelete,
+
+		Schema: map[string]*schema.Schema{
+			"project": &schema.Schema{
```

Comment:
For Xpn Resource, we need to specify both the host project and the service project. Should we call this field "host_project" instead? This would add consistency across both resources and potentially be more explicit for users what project is expected here.

This made me think that maybe having both resource merge into one would be more intuitive:

```hcl
resource "google_compute_shared_vpc" "my-shared-vpc" {
  host_project = "my-host-project"
  service_projects = ["my-service-project1", "my-service-project2"]
}

Looking at enableXpnResource method, it can support potentially other xpn resources than project. If we want to add support for new type of xpn resource, we could easily create a new field. Copying the xpn_resource structure would make it less user friendly IMO. 
See: https://cloud.google.com/compute/docs/reference/latest/projects/enableXpnResource

Reply:
The way I see people using this is the following:

Let's say we have 3 projects. `HOST_ PROJECT`, `SERVICE_PROJECT_1` and `SERVICE_PROJECT_2`.

They have 3 terraform directories.

The first one for the `HOST_PROJECT`:
- terraform service credentials must have compute.XpnAdmin role (or the same permissions through another role)
- sets up the network, firewall rules, etc.
- grants compute.networkUser role to both service projects.
- sets up XPN. Enabling the host and adding the 2 service projects as xpn resources. Since enabling the xpn host and adding xpn resource is likely to be done by the same Terraform config/user, IMO, it makes sense to bundle them up together.

Each service directory:
- Adds the necessary compute/container resources to perform the operations the service is responsible for.
- Since it has compute.networkUser role on the host project, it can read from the host network whatever it needs.

Reply:
Yup, that makes sense. Changed to your proposal, let me know what you think